modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java

```diff
@@ -0,0 +1,318 @@
+		}
+
+		return super.doStartTag();
+	}
+
+	public Integer getActiveDelta() {
+		return _activeDelta;
+	}
+
+	public Integer getActivePage() {
+		return _activePage;
+	}
+
+	public List<PaginationBarDelta> getDeltas() {
+		return _deltas;
+	}
+
+	public List<Integer> getDisabledPages() {
+		return _disabledPages;
+	}
+
+	public Integer getEllipsisBuffer() {
+		return _ellipsisBuffer;
+	}
+
+	public PaginationBarLabels getLabels() {
+		return _labels;
+	}
+
+	public Integer getTotalItems() {
+		return _totalItems;
+	}
+
+	public void setActiveDelta(Integer activeDelta) {
+		_activeDelta = activeDelta;
+	}
+
+	public void setActivePage(Integer activePage) {
+		_activePage = activePage;
+	}
+
+	public void setDeltas(List<PaginationBarDelta> deltas) {
+		_deltas = deltas;
+	}
+
+	public void setDisabledPages(List<Integer> disabledPages) {
+		System.out.println("Disabled pages set");
+		_disabledPages = disabledPages;
+	}
+
+	public void setEllipsisBuffer(Integer ellipsisBuffer) {
+		_ellipsisBuffer = ellipsisBuffer;
+	}
+
+	public void setLabels(PaginationBarLabels labels) {
+		_labels = labels;
+	}
+
+	public void setTotalItems(Integer totalItems) {
+		_totalItems = totalItems;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_activeDelta = null;
+		_activePage = null;
+		_deltas = null;
+		_disabledPages = null;
+		_ellipsisBuffer = null;
+		_labels = null;
+		_totalItems = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/PaginationBar";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("activeDelta", _activeDelta);
+		props.put("activePage", _activePage);
+		props.put("deltas", _deltas);
+		props.put("disabledPages", _disabledPages);
+		props.put("ellipsisBuffer", _ellipsisBuffer);
+		props.put("labels", _labels);
+		props.put("totalItems", _totalItems);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		jspWriter.write("<div class=\"pagination-bar\"><div class=\"");
+		jspWriter.write("dropdown pagination-items-per-page\">");
+
+		ResourceBundle resourceBundle = TagResourceBundleUtil.getResourceBundle(
+			pageContext);
+
+		Integer activeDelta = getActiveDelta();
+
+		if (activeDelta == null) {
+			PaginationBarDelta defaultDelta = getDeltas().get(0);
+
+			Integer label = (Integer)defaultDelta.get("label");
+
+			activeDelta = label;
+		}
+
+		jspWriter.write("<button class=\"dropdown-toggle btn btn-unstyled\"");
+		jspWriter.write(" type=\"button\">");
+		jspWriter.write(activeDelta.toString());
+		jspWriter.write(StringPool.SPACE);
+
+		jspWriter.write(
+			StringUtil.toLowerCase(LanguageUtil.get(resourceBundle, "items")));
+
+		IconTag iconTag = new IconTag();
+
+		iconTag.setSymbol("caret-double-l");
+
+		iconTag.doTag(pageContext);
+
+		if (_labels == null) {
```

Comment:
Aren't We adding this default value on https://github.com/liferay-frontend/liferay-portal/pull/1224/files#diff-60146692a1877389e72f4745a7ee7711d85cc8db0d0a46faa7507e822974faa6R43?

Reply:
Yes, correct. I was initially doing it here (in `processStartTag`) before moving it to `doStartTag` and forgot to remove it.